tests/testsuite/lints.rs

```diff
@@ -0,0 +1,603 @@
+//! Tests for `[lints]`
```

Comment:
There are a couple of missing features (unordered lists) which is why I haven't pushed it since we can't drive towards a new consistency.  That doesn't mean I'm against it just unsure how others would take it.